src/main/kotlin/org/rust/ide/intentions/AddNamespaceIntention.kt

```diff
@@ -0,0 +1,48 @@
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.psi.RsElementTypes.*
+
+class AddNamespaceIntention : RsElementBaseIntentionAction<AddNamespaceIntention.Context>() {
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val (matchExpr, caller, gen, more) = ctx
+        val callWithNamespace =
+            RsPsiFactory(project).createExpression("""${caller.text}::<${gen.text}>${more.text}""") as RsCallExpr
+        matchExpr.replace(callWithNamespace)
+    }
+
+    data class Context(
+        val matchExpr: RsBinaryExpr,
+        val caller: RsExpr,
+        val gen: RsPathExpr,
+        val more: RsParenExpr
+    )
+
+    override fun getText() = "Add missed namespace"
+    override fun getFamilyName() = text
+
+    private fun resolveMatchExpression(element: PsiElement): RsBinaryExpr? {
+        val base = element.parentOfType<RsBinaryExpr>() ?: return null
+        return if (base.left is RsBinaryExpr) {
+                base
+            } else {
```

Comment:
Indentation is a bit messed up. Reformat file (`Ctrl+Alt+l`) should cleanup this.